modules/rules-engine/src/engine.rs

```diff
@@ -54,9 +54,12 @@ impl PulsarEngine {
     }
 
     pub fn process(&self, event: &Event) {
-        self.internal.engine.run(event, |rule| {
-            emit_event(&self.internal.sender, event, &rule.name)
-        })
+        // Run the engine only on non threat events
```

Comment:
I'd specify in the comment why this is needed (eg. "in order not to create loops")